topiary-core/src/atom_collection.rs

```diff
@@ -535,15 +560,38 @@ impl AtomCollection {
             self.atoms.push(Atom::Leaf {
                 content: String::from(node.utf8_text(source)?),
                 id,
-                original_position: node.start_position().into(),
+                original_column: node.start_position().column() as i32,
                 single_line_no_indent: false,
                 multi_line_indent_all: false,
             });
             // Mark all sub-nodes as having this node as a "leaf parent"
             self.mark_leaf_parent(node, node.id());
+            // Test the node for comments
+            // `comments` is sorted in reverse order, so `pop()` gets the first one.
+            while let Some(comment) = comments.pop() {
+                let node_section: InputSection = node.into();
+                // REVIEW: the double borrow is sort of weird, is this idiomatic?
+                if node_section.contains(&(&comment).into()) {
```

Comment:
I can't speak to whether it's idiomatic (@yannham?); however, it makes sense here:

* `comment` is an `AnchoredComment` and you only implement `From<&AnchoredComment> for InputSection`; hence the first reference. You need to use `comment` later, so taking a reference makes sense otherwise you'll get a borrow violation.
* `InputSection::contains`'s argument expects a reference; hence the second. Here, the converted `InputSection` is only used once, so _maybe_ one could implement `From<&AnchoredComment> for &InputSection`, but that would probably require careful lifetime annotations.

My opinion is that the double-`&` is probably a better option than an additional, possibly complex `impl` that's only called once.